src/v2i-hub/SpatPlugin/src/SignalControllerConnection.cpp

```diff
@@ -0,0 +1,78 @@
+/**
+ * Copyright (C) 2024 LEIDOS.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+#include "SignalControllerConnection.h"
+
+namespace SpatPlugin {
+
+    SignalControllerConnection::SignalControllerConnection(const std::string &localIp, unsigned int localPort, const std::string &signalGroupMapping, const std::string &scIp, unsigned int scSNMPPort, const std::string &intersectionName, unsigned int intersectionId) : spatPacketReceiver(std::make_shared<tmx::utils::UdpServer>(localIp, localPort)) ,scSNMPClient(std::make_shared<tmx::utils::snmp_client>(scIp, scSNMPPort ,"administrator", "", "", "")), signalGroupMapping(signalGroupMapping), intersectionName(intersectionName), intersectionId(intersectionId) {
+
+    };
+    bool SignalControllerConnection::initializeSignalControllerConnection() const {
+        // TODO : Set Intersection ID and  Intersection Name
+        tmx::utils::snmp_response_obj resp;
+        resp.val_int = 2;
+        resp.type = tmx::utils::snmp_response_obj::response_type::INTEGER;
+        return scSNMPClient->process_snmp_request(NTCIP1202V3::ENABLE_SPAT_OID, tmx::utils::request_type::SET, resp);
+    };
+
+    void SignalControllerConnection::receiveBinarySPAT(const std::shared_ptr<SPAT> &spat, uint64_t timeMs ) const {
+        FILE_LOG(tmx::utils::logDEBUG) << "Receiving binary SPAT ..." << std::endl;
+        char buf[1000];
```

Comment:
This is definitely outside the scope for my issue. This buffer was allocated like this in the current implementation. Changing these values and retesting is definitely not in scope for this bug fix for simulation.

Reply:
It might be a misunderstanding. I meant use a constant or a variable that has value 1000 rather than just 1000. It seems the three 1000 values mean both buffer_size and time_out. It is for code readability. Not sure if that has significant on regression testing.